File: Semester 4/Programming and Projection Enironments/mpp-proiect-csharp/Proiect_MPP/Proiect_MPP/repository/people/IPersonRepository.cs
using Proiect_MPP.domain.people;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Proiect_MPP.repository.people
{
    public interface IPersonRepository<T> : Repository<T, int> where T : Person
    {
        List<T> getPersonByFirstName(string firstName);
        List<T> getPersonByLastName(string lastName);
    }
}

File: Semester 4/Programming and Projection Enironments/mpp-proiect-java/Proiect_MPP/Thrift/src/main/java/root/proiect_mpp/repositories/people/IPersonRepository.java
package root.proiect_mpp.repositories.people;

import root.proiect_mpp.domain.people.Person;
import root.proiect_mpp.repositories.Repository;

import java.util.Collection;

public interface IPersonRepository<T extends Person> extends Repository<T, Integer> {
    Collection<T> getPersonByFirstName(String firstName);
    Collection<T> getPersonByLastName(String lastName);
}
